src/reconcile/rec_write.c

```diff
@@ -2473,89 +2457,59 @@ __rec_split_grow(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t add_len)
 }
 
 /*
- * __rec_split_write_prev_and_shift_cur --
- *	Write the previous split chunk to the disk as a page. Shift the contents
- *	of the current chunk to the start of the buffer, making space for a new
- *	chunk to be written.
- *	If the caller asks for a chunk resizing, the boundary between the two
- *	chunks is readjusted to the minimum split size boundary details stored
- *	in the previous chunk, letting the current chunk grow at the cost of the
- *	previous chunk.
+ * __rec_split_write_prev_and_swap_buf --
+ *	If there is a previous split chunk held in the memory, write it to the
+ *	disk as a page. If there isn't one, initialize a second buffer to hold
+ *	it. Swap the previous and current buffer pointers, making the current
+ *	buffer as the previous one.
  */
 static int
-__rec_split_write_prev_and_shift_cur(
-    WT_SESSION_IMPL *session, WT_RECONCILE *r, bool resize_chunks)
+__rec_split_write_prev_and_swap_buf(
+    WT_SESSION_IMPL *session, WT_RECONCILE *r)
 {
-	WT_BM *bm;
 	WT_BOUNDARY *bnd_cur, *bnd_prev;
-	WT_BTREE *btree;
-	WT_PAGE_HEADER *dsk, *dsk_tmp;
-	size_t cur_len, len;
-	uint8_t *dsk_start;
-
-	WT_ASSERT(session, r->bnd_next != 0);
```

Comment:
r->bnd_next starts initialised as 0 and r->prev_img_ptr starts initialised as NULL. bnd_next is index into the boundary array, we increment it when we create a boundary. To create the first boundary, first we initialise the second buffer, let prev_img_ptr point to the first image and current image pointer point to the second image, then we increment bnd_next to 1.

So assert is right in checking "r->prev_img_ptr != NULL && r->bnd_next != 0".
Said that I am updating assert to check after first boundary creation.

Assert will check the following now:

        WT_ASSERT(session, (r->prev_img_ptr == NULL && r->bnd_next == 0) ||      
            (r->prev_img_ptr != NULL && r->bnd_next != 0));